### ISSUE TITLE
Cancel button doesn't work on non-posts after switching languages

### ISSUE THREAD

**Issue by LouisStAmour** [refinery/refinerycms#2915]
Similar to issue #2044 which was created (and fixed) only for pages, I found the cancel button wasn't working for me when switching languages, as the referring URL is prefered over history navigation via JS (and either way, it's incorrect, unless maybe PJAX...)

The problem is that fixing it for pages in its form partial doesn't help much for custom created content types or any other uses of this partial: https://github.com/refinery/refinerycms/blob/master/core/app/views/refinery/admin/_form_actions.html.erb#L16

Partly the problem is that it, like many other page _form customizations, were left out of https://github.com/refinery/refinerycms/blob/master/core/lib/generators/refinery/engine/templates/app/views/refinery/namespace/admin/plural_name/_form.html.erb#L51 but part of it also is that ... now I have to go and copy and paste this line across a dozen content types' _form templates?! Ick. :)

Just thought I'd report this, as it's too late in the evening for me to compose a PR for it.

**Comment by parndt** [comment]
@LouisStAmour are you still facing this issue / did you come up with a patch? :smile:

**Comment by parndt** [comment]
Any update?

**Comment by bricesanchez** [comment]
So if i understand well , we need this to fix the problem? 

```
# https://github.com/refinery/refinerycms/blob/master/core/lib/generators/refinery/engine/templates/app/views/refinery/namespace/admin/plural_name/_form.html.erb

[...]

  <%%= render '/refinery/admin/form_actions', :f => f,
             :continue_editing => false,
             :delete_title => t('delete', :scope => 'refinery.<%= plural_name %>.admin.<%= plural_name %>.<%= singular_name %>'),
             :delete_confirmation => t('message', :scope => 'refinery.admin.delete'<% if (title = attributes.detect { |a| a.type.to_s == "string" }).present? %>, :title => @<%= singular_name %>.<%= title.name %><% end %>),
cancel_url => refinery.admin_<%= plural_name %>_path -%>

[...]
```

**Comment by bricesanchez** [comment]
This is fixed in #3351
